interfaces/apparmor/backend.go

```diff
@@ -103,7 +103,26 @@ func (b *Backend) Setup(snapInfo *snap.Info, developerMode bool, repo *interface
 	if errReload != nil {
 		return errReload
 	}
-	return errUnload
+	if errUnload != nil {
+		return errUnload
+	}
+	// Sanity checking, see if all the profiles are really loaded and load
+	// those profiles that are not. This can happen if the files were on disk
+	// but were not really loaded (e.g. due to an interrupted install).
```

Comment:
I previously mentioned that I had a hard time understanding the purpose of this addition. We are invoking the parser in such a way that you point it at a policy file, it compiles it into a cache file and loads it into the kernel. If the cache file doesn't exist on reboot, the parser will create it; if it does exist and is newer than the profile, the parser will use the cache file; if it does exist but is older than the profile, the parser will recompile and create a new cache file. Snappy should continue with this approach.

Therefore, all you need to do is make sure all the profiles you want are created and then run the parser on them (obviously checking for errors; but also don't update the mtime unless they actually changed). If things get interrupted on what would otherwise be a successful parser run, don't invalidate the cache files you already computed, just run the parser on them in the way you always would (reloading an already loaded profile from cache is cheap, fast and robust).